public/components/datasources/components/manage/associated_objects/associated_objects_tab.tsx

```diff
@@ -358,20 +482,53 @@ export const AssociatedObjectsTab: React.FC<AssociatedObjectsTabProps> = ({
       <EuiPanel>
         <AssociatedObjectsHeader />
         <EuiHorizontalRule />
-        <AccelerationsRecommendationCallout />
-        <EuiSpacer />
-        {associatedObjects.length > 0 ? (
-          <EuiInMemoryTable
-            items={filteredObjects}
-            columns={columns}
-            search={search}
-            pagination={pagination}
-            sorting={sorting}
-            noItemsMessage={associatedObjects.length === 0 ? noDataMessage : undefined}
-            data-test-subj={ASSC_OBJ_TABLE_SUBJ}
-          />
+        {isFirstTimeLoading ? (
+          <AssociatedObjectsTabLoading objectType="databases" warningMessage={false} />
         ) : (
-          noDataMessage
+          <>
+            {databasesLoadStatus === DirectQueryLoadingStatus.SUCCESS &&
+            cachedDatabases.length === 0 ? (
+              <AssociatedObjectsTabEmpty cacheType="databases" />
+            ) : (
+              <>
+                <AccelerationsRecommendationCallout />
+                <EuiSpacer />
+                <EuiFlexGroup direction="row">
+                  <EuiFlexItem grow={false}>
+                    <EuiSelectable
+                      searchable={true}
+                      singleSelection={'always'}
+                      searchProps={{ placeholder: 'Search for databases' }}
```

Comment:
We have 3 ternary operators here. We should simplify these to make it more readable. There are many ways to do it: 
* Split into smaller components 
* Or can simplify the conditions for rendering